frontend/src/components/Public.vue

```diff
@@ -73,8 +64,7 @@ export default Vue.extend({
     const ls = UserLocationService.getInstance();
 
     const a = new InfoService();
-    this.$store.dispatch('grabRoutes');
-    this.$store.dispatch('grabStops');
+    this.$store.dispatch('grabStops').then(() => this.$store.dispatch('grabRoutes'));
```

Comment:
I don't like stacking promises like this, you could probably acheive something similar with Promise.all() https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all and ensuring your are doing undefined/null value checks where necessary